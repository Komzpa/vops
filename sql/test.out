CREATE EXTENSION
CREATE TABLE
CREATE TABLE
INSERT 0 6
 populate 
----------
 
(1 row)

 unnest 
--------
 (2)
 (3)
 (4)
(3 rows)

 countall 
----------
        4
(1 row)

 count 
-------
    60
(1 row)

 count 
-------
     4
(1 row)

 count | count | sum | avg | min | max |     variance     | var_pop |     var_samp     |      stddev      |    stddev_pop    |   stddev_samp    
-------+-------+-----+-----+-----+-----+------------------+---------+------------------+------------------+------------------+------------------
     4 |     4 |  10 | 2.5 |   1 |   4 | 1.66666666666667 |    1.25 | 1.66666666666667 | 1.29099444873581 | 1.11803398874989 | 1.29099444873581
(1 row)

 count | count | sum | avg | min | max |     variance     | var_pop |     var_samp     |      stddev      |    stddev_pop    |   stddev_samp    
-------+-------+-----+-----+-----+-----+------------------+---------+------------------+------------------+------------------+------------------
     4 |     4 |  10 | 2.5 |   1 |   4 | 1.66666666666667 |    1.25 | 1.66666666666667 | 1.29099444873581 | 1.11803398874989 | 1.29099444873581
(1 row)

 count | count | sum | avg | min | max | variance |      var_pop      | var_samp | stddev |    stddev_pop     | stddev_samp 
-------+-------+-----+-----+-----+-----+----------+-------------------+----------+--------+-------------------+-------------
     3 |     3 |   9 |   3 |   2 |   4 |        1 | 0.666666666666667 |        1 |      1 | 0.816496580927726 |           1
(1 row)

 count | count | sum | avg | min | max | variance |      var_pop      | var_samp | stddev |    stddev_pop     | stddev_samp 
-------+-------+-----+-----+-----+-----+----------+-------------------+----------+--------+-------------------+-------------
     3 |     3 |   9 |   3 |   2 |   4 |        1 | 0.666666666666667 |        1 |      1 | 0.816496580927726 |           1
(1 row)

