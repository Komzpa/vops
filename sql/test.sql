create extension vops;
create table s(x real);
create table v(x vops_float4);
insert into s values(1.0),(2.0),(null),(3.0),(null),(4.0);
select populate(destination:='v'::regclass, source:='s'::regclass);
select unnest(v.*) from v where x > 1;
select countall(*) from v where x is not null;
select count(*) from v where x is null;
select count(*) from v where x is not null;
select count(*),count(x),sum(x),avg(x),min(x),max(x),variance(x),var_pop(x),var_samp(x),stddev(x),stddev_pop(x),stddev_samp(x) from v where x >= 0.0;
select count(*),count(x),sum(x),avg(x),min(x),max(x),variance(x),var_pop(x),var_samp(x),stddev(x),stddev_pop(x),stddev_samp(x) from s where x >= 0.0;
select count(*),count(x),sum(x),avg(x),min(x),max(x),variance(x),var_pop(x),var_samp(x),stddev(x),stddev_pop(x),stddev_samp(x) from v where x > 1.0;
select count(*),count(x),sum(x),avg(x),min(x),max(x),variance(x),var_pop(x),var_samp(x),stddev(x),stddev_pop(x),stddev_samp(x) from s where x > 1.0;
select count(*) from v where ifnull(x, 0) >= 0;
