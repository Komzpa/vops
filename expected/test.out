create extension vops;
create table s(x real);
create table v(x vops_float4);
insert into s values(1.0),(2.0),(null),(3.0),(null),(4.0);
select populate(destination:='v'::regclass, source:='s'::regclass);
 populate 
----------
 
(1 row)

select unnest(v.*) from v where x > 1;
 unnest 
--------
 (2)
 (3)
 (4)
(3 rows)

select countall(*) from v where x is not null;
 countall 
----------
        4
(1 row)

select count(*) from v where x is null;
 count 
-------
    60
(1 row)

select count(*) from v where x is not null;
 count 
-------
     4
(1 row)

select count(*),count(x),sum(x),avg(x),min(x),max(x),variance(x),var_pop(x),var_samp(x),stddev(x),stddev_pop(x),stddev_samp(x) from v where x >= 0.0;
 count | count | sum | avg | min | max |     variance     | var_pop |     var_samp     |      stddev      |    stddev_pop    |   stddev_samp    
-------+-------+-----+-----+-----+-----+------------------+---------+------------------+------------------+------------------+------------------
     4 |     4 |  10 | 2.5 |   1 |   4 | 1.66666666666667 |    1.25 | 1.66666666666667 | 1.29099444873581 | 1.11803398874989 | 1.29099444873581
(1 row)

select count(*),count(x),sum(x),avg(x),min(x),max(x),variance(x),var_pop(x),var_samp(x),stddev(x),stddev_pop(x),stddev_samp(x) from s where x >= 0.0;
 count | count | sum | avg | min | max |     variance     | var_pop |     var_samp     |      stddev      |    stddev_pop    |   stddev_samp    
-------+-------+-----+-----+-----+-----+------------------+---------+------------------+------------------+------------------+------------------
     4 |     4 |  10 | 2.5 |   1 |   4 | 1.66666666666667 |    1.25 | 1.66666666666667 | 1.29099444873581 | 1.11803398874989 | 1.29099444873581
(1 row)

select count(*),count(x),sum(x),avg(x),min(x),max(x),variance(x),var_pop(x),var_samp(x),stddev(x),stddev_pop(x),stddev_samp(x) from v where x > 1.0;
 count | count | sum | avg | min | max | variance |      var_pop      | var_samp | stddev |    stddev_pop     | stddev_samp 
-------+-------+-----+-----+-----+-----+----------+-------------------+----------+--------+-------------------+-------------
     3 |     3 |   9 |   3 |   2 |   4 |        1 | 0.666666666666667 |        1 |      1 | 0.816496580927726 |           1
(1 row)

select count(*),count(x),sum(x),avg(x),min(x),max(x),variance(x),var_pop(x),var_samp(x),stddev(x),stddev_pop(x),stddev_samp(x) from s where x > 1.0;
 count | count | sum | avg | min | max | variance |      var_pop      | var_samp | stddev |    stddev_pop     | stddev_samp 
-------+-------+-----+-----+-----+-----+----------+-------------------+----------+--------+-------------------+-------------
     3 |     3 |   9 |   3 |   2 |   4 |        1 | 0.666666666666667 |        1 |      1 | 0.816496580927726 |           1
(1 row)

select count(*) from v where ifnull(x, 0) >= 0;
 count 
-------
    64
(1 row)

